R/util_data_summaries.R

```diff
@@ -126,3 +155,509 @@ ds.summarize_by_column <- function(
+    if( sum( ! predictors %in% varlist ) > 0){
+        varlist$covs <- predictors[ ! predictors %in% varlist ]
+    }
+    
+    return(varlist)
+}
+
+
+ds.glm1 <- function( formula, data, mod_family=NULL ){
+    # run the appropriate model, unless mod_family override is supplied
+    dv_variable <- ds.helper$glm1_formula_to_varlist(formula)[["dv"]]
+    if( is.null(mod_family) ){
+        dv_var_type <- ds.helper$variable_type(data[,dv_variable])
+        mod_family  <- util.recode(
+                            dv_var_type,
+                            names(ds.helper$map_dv_to_glm_family),
+                            unlist(ds.helper$map_dv_to_glm_family)
+                        )
+    }
+    if( ! mod_family %in% ds.helper$map_dv_to_glm_family ){
+        util.warn("Unsupported model family requested: " %+% mod_family )
+        mod_obj <- NULL
+    } else{
+        mod_obj <- glm( formula , data, family=mod_family )
+    }
+    return(mod_obj)
+}
+
+ds.glm1s <- function( formulas, data, family=NULL ){
+    # run each formula on data.frame supplied in data argument
+    # use the outcome type to select the model, or override with family
+    models <- list()
+    if( is.null( family ) ){
+        for( i in 1:length(formulas)){
+            models[[i]] <- ds.glm1( formulas[i], data=data )
+        }        
+    } else{
+        for( i in 1:length(formulas)){
+            models[[i]] <- ds.glm1( formulas[i], data=data, family=family )
+        }    
+    }
+    return(models)
+}
+
+ds.summarize_glm1 <- function( glm_model ){
+    # extracts frequently used summary statistics from glm_model
+    # glm_model must adhere to ds.glm1 spec, 
+    # see ds.validate_glm1_model
+    
+    if( ! ds.validate_glm1_model(glm_model)){
+        # glm1 only supports numerical or boolean IVs and MODs
+        # others will issue a warning
+        return(list())
+    }
+    
+    msl <- list()  # msl = "model summary list"
+    varlist <- ds.helper$glm1_formula_to_varlist(glm_model$formula)
+    
+    # populate the feature list
+    msl$formula <- glm_model$formula
+    msl$dv      <- varlist$dv
+    msl$family  <- glm_model$family$family
+    msl$n   <- sum(!util.is_blank(glm_model$y))    
+    msl$df  <- glm_model$df.residual
+    msl$aic <- glm_model$aic
+    msl$iv     <- varlist$iv
+    msl$mod    <- varlist$mod
+    msl$covs   <- paste(varlist$covs, collapse=", ")
+    msl$dv_mean   <- mean(glm_model$y, na.rm=TRUE)
+    
+    # extract coefficients
+    coefs <- summary(glm_model)$coefficients
+    
+    # use of column and row numbers seems dangerous;
+    # however order should be consistent when 
+    # ds.glm1 spec is used
+    # also, irregularities between different model types
+    # e.g., t vs z for guassian and binomial families
+    # and level is appended to row.names when TRUE/FALSE
+    msl$iv_coef <- coefs[2,1]
+    msl$iv_se <- coefs[2,2]
+    msl$iv_stat <- coefs[2,3]
+    msl$iv_p <- coefs[2,4]
```

Comment:
Here's a place where the purrr package could offer more flexibility and make it easier to e.g., extract coefficient info for multiple levels of a categorical dv